src/pudl/transform/ferc1.py

```diff
@@ -853,6 +853,533 @@ def strip_col_of_non_ints(self, col: pd.Series) -> pd.DataFrame:
+        out_df.loc[
+            (not_hydro & ~obvious_license)
+            | not_license
+            | (year_vs_num & ~obvious_license & ~exceptions),
+            "license_id_ferc1",
+        ] = np.nan
+        # Fill fuel type with hydro
+        out_df.loc[
+            out_df["license_id_ferc1"].notna() & (out_df["fuel_type"] == "other"),
+            "fuel_type",
+        ] = "hydro"
+
+        return out_df
+
+    def label_row_type(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Label rows as headers, notes, or totals."""
+        # Define header qualifications
+        header_strings = [
+            "hydro",
+            "hyrdo",
+            "internal",
+            "wind",
+            "solar",
+            "gas",
+            "diesel",
+            "diesal",
+            "steam",
+            "other",
+            "combustion",
+            "combustine",
+            "fuel cell",
+            "hydraulic",
+            "waste",
+            "landfill",
+            "photovoltaic",
+            "nuclear",
+            "oil",
+            "renewable",
+            "facilities",
+            "combined cycle",
+        ]
+        nonheader_strings = [
+            "#",
+            r"\*",
+            "pg",
+            "solargenix",
+            "solargennix",
+            r"\@",
+            "rockton",
+            "albany steam",
+        ]
+        exceptions = [
+            "hydro plants: licensed proj. no.",
+            "hydro license no.",
+            "hydro: license no.",
+            "hydro plants: licensed proj no.",
+        ]
+        not_header_if_cols_na = [
+            "construction_year",
+            "net_generation_mwh",
+            "total_cost_of_plant",
+            "capex_per_mw",
+            "opex_total",
+            "opex_fuel",
+            "opex_maintenance",
+            "fuel_cost_per_mmbtu",
+        ]
+
+        def label_header_rows(df: pd.DataFrame) -> pd.DataFrame:
+            """Label header rows.
+
+            This function labels rows it believes are possible headers based on whether
+            they contain information in certain key columns. Of those possible headers,
+            ones that contain a specific key word or phrase are dubbed headers. Leftover
+            possible header rows are evaluated as possible note rows in the
+            _label_notes_rows() function.
+            """
+            logger.info(f"{self.table_id.value}: Labeling header rows")
+
+            # Label possible header rows (based on the nan cols specified above)
+            df.loc[
+                df.filter(not_header_if_cols_na).isna().all(1), "possible_header"
+            ] = True
+
+            # Label good header rows (based on whether they contain key strings)
+            possible_header = df["possible_header"]
+            good_header = df["plant_name_ferc1"].str.contains("|".join(header_strings))
+            not_bad = ~df["plant_name_ferc1"].str.contains("|".join(nonheader_strings))
```

Comment:
I think it should be `not_bad_header`. Like there are some exceptions that might otherwise get cast as a header that should not.

Reply:
does that mean its not a header? or its not a good header?

Reply:
It's supposed to be a double negative so there is "good" and then "not bad" as in, there are some exceptions codified in the "not bad" group that would otherwise be considered bad headers. Maybe it would be clearer to just change it to "bad_header" or "not_header" and then add the ~ in the call like so: `df.loc[good_header, ~bad_header]`